src/sagemaker_containers/worker.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License'). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the 'license' file accompanying this file. This file is
+# distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import collections
+
+from flask import Flask, Response
+
+import sagemaker_containers as smc
+
+env = smc.environment.ServingEnvironment()
+
+
+def default_healthcheck_fn():  # type: () -> Response
+    """Ping is default health-check handler. Returns 200 with no content.
+
+    During a new serving container startup, Amazon SageMaker starts sending periodic GET requests to the /ping endpoint
+    to ensure that the container is ready for predictions.
+
+    The simplest requirement on the container is to respond with an HTTP 200 status code and an empty body. This
+    indicates to Amazon SageMaker that the container is ready to accept inference requests at the /invocations endpoint.
+
+    If the container does not begin to consistently respond with 200s during the first 30 seconds after startup,
+    the CreateEndPoint and UpdateEndpoint APIs will fail.
+
+    While the minimum bar is for the container to return a static 200, a container developer can use this functionality
+    to perform deeper checks. The request timeout on /ping attempts is 2 seconds.
+
+    More information on how health-check works can be found here:
+    https://docs.aws.amazon.com/sagemaker/latest/dg/your-algorithms-inference-code.html#your-algorithms-inference-algo-ping-requests
+
+    Returns:
+        (flask.Response): with status code 200
+    """
+    return Response(status=smc.status_codes.OK)
+
+
+def run(transform_fn, initialize_fn=None, healthcheck_fn=None, module_name=None):
+    # type: (function, function or None, str or None) -> Flask
+    """Creates and Flask application from a transformer.
+
+    Args:
+        transform_fn (function): responsible to make predictions against the model. Follows the signature:
+
+            * Returns:
+                `sagemaker_containers.worker.TransformSpec`: named tuple with prediction data.
+
+
+        initialize_fn (function, optional): this function is called when the Flask application starts.
```

Comment:
Should this default to the function that loads the model?
I want to avoid this case: my model takes a long time to load, but the server starts, so ping returns false 200s (when it can't accept inference requests yet). I think the default behavior should be to delay default_healthcheck_fn from returning 200s until the model is loaded

Reply:
A large model can take longer to load than then that time limit defined by SageMaker hosting to receive a successful ping (health check).  This is an issue already in the roadmap.